wft4galaxy/common.py

```diff
@@ -318,3 +270,91 @@ def unload_workflows(self):
             raise RuntimeError("WorkflowLoader not initialized")
         for _, wf in _iteritems(self._workflows):
             self.unload_workflow(wf.id)
+
+
+# GalaxyInstance wrapper
+class GalaxyInstance(ObjGalaxyInstance):
+    def __init__(self, url, api_key=None, email=None, password=None,
+                 max_retries=MAX_RETRIES, retry_delay=RETRY_DELAY, polling_interval=POLLING_INTERVAL):
+        super(GalaxyInstance, self).__init__(url, api_key, email, password)
+        if max_retries is not None:
+            self.max_retries = max_retries
+        if retry_delay is not None:
+            self.retry_delay = retry_delay
+        if polling_interval is not None:
+            self.polling_interval = polling_interval
+
+    @property
+    def max_retries(self):
+        return self.gi.max_get_attempts
+
+    @max_retries.setter
+    def max_retries(self, v):
+        self.gi.max_get_attempts = v
+
+    @property
+    def retry_delay(self):
+        return self.gi.get_retry_delay
+
+    @retry_delay.setter
+    def retry_delay(self, v):
+        self.gi.get_retry_delay = v
+
+    @property
+    def polling_interval(self):
+        return self._polling_interval
+
+    @polling_interval.setter
+    def polling_interval(self, interval):
+        self._polling_interval = interval
+
+
+def configure_env_galaxy_server_instance(config, options, base_config=None):
+    config["galaxy_url"] = options.galaxy_url \
```

Comment:
Interesting construct... what is the operator precedence in these cases? Found it a bit hard to read.

Reply:
It's the usual operator precedence, i.e., `and` has an higher precedence than `or`.
In this case you have something like `A or B or C`, where `B = X and Y`